mage/packages.go

```diff
@@ -0,0 +1,982 @@
+	defer func() { m.Complete(err) }()
+
+	fioImage := fmt.Sprint("fio:", fioTag)
+
+	err = m.DockerBuild().
+		SetBuildArg("BUILD_BOX", buildBoxName()).
+		SetBuildArg("FIO_BRANCH", fioTag).
+		SetPull(false).
+		AddTag(fioImage).
+		Build(context.TODO(), "assets/fio")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	wd, _ := os.Getwd()
+
+	err = m.DockerRun().
+		SetRemove(true).
+		AddVolume(magnet.DockerBindMount{
+			Source:      filepath.Join(wd, "build/"),
+			Destination: "/local",
+		}).
+		Run(context.TODO(), fioImage, "cp", "/gopath/native/fio/fio", filepath.Join("/local", "fio"))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	defer os.Remove("build/fio")
+
+	return trace.Wrap(pkgFio.ImportPackage(m, "build/fio"))
+}
+
+func (Package) Selinux() (err error) {
+	mg.Deps(Build.Go)
+
+	m := root.Target("package:selinux")
+	defer func() { m.Complete(err) }()
+
+	tmpDir, err := ioutil.TempDir("", "build-selinux")
+	if err != nil {
+		return trace.ConvertSystemError(err)
+	}
+
+	defer os.RemoveAll(tmpDir)
+
+	_, err = m.Exec().SetWD(tmpDir).Run(context.TODO(),
+		"git",
+		"clone",
+		selinuxRepo,
+		"--branch", selinuxBranch,
+		"--depth=1",
+		"./",
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = m.Exec().SetWD(tmpDir).Run(context.TODO(),
+		"git",
+		"submodule",
+		"update",
+		"--init",
+		"--recursive",
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = m.Exec().SetWD(tmpDir).
+		Run(context.TODO(), "make", "BUILDBOX_INSTANCE=")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	cachePath := filepath.Join("build/apps", fmt.Sprint("selinux.", pkgSelinux.version, ".tar.gz"))
+
+	_, err = m.Exec().SetWD(filepath.Join(tmpDir, "selinux/output")).
+		Run(context.TODO(),
+			"tar",
+			"czf", cachePath,
+			"gravity.pp.bz2", "container.pp.bz2", "gravity.statedir.fc.template",
+		)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(pkgFio.ImportPackage(m, cachePath))
+}
+
+func (Package) Planet() (err error) {
+	mg.Deps(Build.Go)
+
+	m := root.Target("package:planet")
+	defer func() { m.Complete(err) }()
+
+	packageList, err := magnet.Output(context.TODO(),
+		consistentGravityBin(),
+		"--state-dir", consistentStateDir(),
+		"package",
+		"list",
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if strings.Contains(packageList, pkgPlanet.Locator()) {
+		m.SetCached(true)
+		return nil
+	}
+
+	if _, err := os.Stat(pkgPlanet.DefaultCachePath()); !os.IsNotExist(err) {
+		m.SetCached(true)
+		return trace.Wrap(pkgPlanet.ImportPackage(m, pkgPlanet.DefaultCachePath()))
+	}
+
+	err = pkgPlanet.BuildApp()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(pkgPlanet.ImportPackage(m, pkgPlanet.DefaultCachePath()))
+}
+
+func (Package) Web() (err error) {
+	mg.Deps(Build.Go)
+
+	m := root.Target("package:web-assets")
+	defer func() { m.Complete(err) }()
+
+	webImage := fmt.Sprint("telekube-oss-web:", pkgWebAssets.version)
+	contextPath := "web/"
+	if enterprise != "" {
```

Comment:
I'm curious how hard it would be (or even worth the effort) to try and move enterprise-specific logic to e/ submodule so open-source version doesn't have enterprise-specific code.

Reply:
I'm not sure what benefit that gets us?